dashboard/src/components/IasqlEditor.tsx

```diff
@@ -130,39 +135,19 @@ export default function IasqlEditor() {
     });
   }, []);
 
-  // Set up editor autocomplete settings
   useEffect(() => {
-    // https://stackoverflow.com/questions/53622096/how-to-specify-a-list-of-custom-tokens-to-list-for-autocompletion-in-ace-editor
-    editorRef?.current?.editor?.completers?.push({
-      getCompletions: (_editor: any, _session: any, _pos: any, _prefix: any, callback: any) => {
-        const completions: any[] = [];
-        // we can use session and pos here to decide what we are going to show
-        const autoCompleteIasqlKeywords = [
-          // Table Names
-          ...Object.values(installedModules)
-            .map(mod => Object.keys(mod))
-            .flat()
-            .map(value => ({ value, meta: 'Table' })),
-          // Column Names
-          ...Object.values(installedModules)
-            .map(mod => Object.values(mod))
-            .flat()
-            .map(tbl => Object.entries(tbl).map(([value, meta]) => ({ value, meta })))
-            .flat(),
-          // Function Names
-          ...Object.values(functions)
-            .map(fn => Object.keys(fn))
-            .flat()
-            .map(fn => ({ value: fn, meta: 'Function' })),
-        ];
-        autoCompleteIasqlKeywords?.push({ value: 'iasql_help()', meta: 'Function' });
-        autoCompleteIasqlKeywords?.forEach(completion => {
-          completions.push(completion);
-        });
-        callback(null, completions);
-      },
-    });
-  }, [installedModules, functions]);
+    const completer = {
+      getCompletions: async function(editor: any, session: any, pos: any, prefix: any, callback: any) {
+        console.log("in get completions");
+        console.log(suggestions);
+        if (suggestions) {
+          callback(null, suggestions);
+        }
+        else callback(null, []);
+      }
+    }
+    if (editorRef.current?.editor.completers) editorRef.current.editor.completers = [completer];
```

Comment:
```suggestion
    editorRef.current.editor.completers = [completer];
```

Reply:
this actually needs the conditional or fails due to null safety checks